fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java

```diff
@@ -53,6 +53,12 @@ public static Type getCommonSuperType(Type t1, Type t2) {
         if (t1.isStructType() && t2.isStructType()) {
             return getCommonStructType((StructType) t1, (StructType) t2);
         }
+        if (t1.isBoolean() && t2.isComplexType()) {
+            return t2;
+        }
+        if (t1.isComplexType() && t2.isBoolean()) {
+            return t1;
+        }
```

Comment:
what about the real bool (not from null) with complex types? may be need to disable it.

prefer writing to below format.

Reply:
emm, it's speical, but we will handle TYPE_NULL to TYPE_BOOLEAN, so we need handle TYPE_BOOLEAN

Reply:
like: map{null: null}

we can't deriver the map's type is MAP<NULL, NULL>, it's will cause so many question, we need modify the type to a primary type, like MAP<BOOLEAN, BOOLEAN>

so, in some functions, like: `cast(map{null: null} as MAP<int, MAP<INT, INT>>)`, we will meet BOOLEAN cast to MAP<INT, INT>